src/tools/rbd/action/Export.cc

```diff
@@ -564,10 +575,21 @@ static int do_export(librbd::Image& image, const char *path, bool no_progress,
   int fd;
   int max_concurrent_ops = g_conf().get_val<uint64_t>("rbd_concurrent_management_ops");
   bool to_stdout = (strcmp(path, "-") == 0);
+  bool is_blk_dev = false;
   if (to_stdout) {
     fd = STDOUT_FILENO;
   } else {
-    fd = open(path, O_WRONLY | O_CREAT | O_EXCL | O_BINARY, 0644);
+    // On Windows, we need read access in order to retrieve the block device size.
+    int flags = O_RDWR | O_BINARY | O_EXCL;
+    is_blk_dev = utils::is_blk_dev(path);
+    if (!is_blk_dev) {
+      flags |= O_CREAT;
+    } else if (export_format != 1) {
+      cerr << "rbd: exporting to raw block devices is only allowed "
+           << "using the v1 image format" << std::endl;
```

Comment:
Thinking about those checks on the import side (see other comments), I wonder if this is unnecessarily restrictive.  For `--export-format 2`, the image size is recorded in the blob, so the size of the block device isn't _that_ relevant.

Perhaps instead of `if (export_format != 1) { bail }`, it would be better to go with `if (export_format == 1) { perform size check }`?

Reply:
Ok, now I remember. The main issue with v2 was the fact that Windows forces the writes to be sector aligned, so we decided to only support v1 for the time being and fail early with a meaningful error message for v2:

https://github.com/ceph/ceph/pull/49358#discussion_r1054354190

Reply:
Even reads must be sector aligned, so v2 imports fail while retrieving the image banner:

```
 ./rbd.exe import --export-format 2 \\.\PhysicalDrive5 rbd_v2_import
rbd: failed to decode diff banner
rbd: import header failed.
Importing image: 0% complete...failed.
rbd: import failed: (22) Invalid argument
```

The same operation worked fine on Linux:

```
sudo ./bin/rbd import --export-format 2 /dev/rbd0 rbd_v2_import
Importing image: 100% complete...done.
```

That being considered, I guess we can keep those image format checks, but only on Windows.